Can't find UART with 6.0.0-a5

### Linux Distribution

```
$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.1 LTS
Release:        24.04
Codename:       noble

$ uname -a
Linux aml-s805x-ac 6.1.92-15907-gf36fd2695db3 #1 SMP PREEMPT_DYNAMIC Wed Jun  5 02:21:06 EDT 2024 aarch64 aarch64 aarch64 GNU/Linux
```

### What happened

from v6.0.0-a1
```
###### Identifying MCU possibly connected via UART ...
/bin/sh: 1: Syntax error: word unexpected (expecting ")")
[ERROR] Unable to find a UART device!
Command '"find /dev -maxdepth 1 -regextype posix-extended -regex "^\/dev\/tty(AMA0|S0)$" 2>/dev/null"' returned non-zero exit status 2.
```

### What did you expect to happen

from v5.1.1
```
###### Identifying MCU possibly connected via UART ...

 ● MCU #1: /dev/ttyS0
```

### How to reproduce

connect USB serial adapter
start kiauh
select v6 alpha
select `4) [Advanced] `
select `4) [Get MCU ID] `
select `2) UART `


### Additional information

option `3) USB (DFU mode) ` in 6.0.0-a1 also prints errors rather than just saying `No MCU found!` as v5.1.1 does, not sure if it is actually working but noisy or if it would also have failed to detect something if connected.

--- a/kiauh/components/klipper_firmware/firmware_utils.py
+++ b/kiauh/components/klipper_firmware/firmware_utils.py
@@ -6,7 +6,7 @@
 #                                                                         #
 #  This file may be distributed under the terms of the GNU GPLv3 license  #
 # ======================================================================= #
-
+import re
 from subprocess import (
     DEVNULL,
     PIPE,
@@ -50,8 +50,8 @@ def find_firmware_file() -> bool:
 
 def find_usb_device_by_id() -> List[str]:
     try:
-        command = "find /dev/serial/by-id/* 2>/dev/null"
-        output = check_output(command, shell=True, text=True)
+        command = "find /dev/serial/by-id/*"
+        output = check_output(command, shell=True, text=True, stderr=DEVNULL)
         return output.splitlines()
     except CalledProcessError as e:
         Logger.print_error("Unable to find a USB device!")
@@ -61,9 +61,14 @@ def find_usb_device_by_id() -> List[str]:
 
 def find_uart_device() -> List[str]:
     try:
-        command = '"find /dev -maxdepth 1 -regextype posix-extended -regex "^\/dev\/tty(AMA0|S0)$" 2>/dev/null"'
-        output = check_output(command, shell=True, text=True)
-        return output.splitlines()
+        cmd = "find /dev -maxdepth 1"
+        output = check_output(cmd, shell=True, text=True, stderr=DEVNULL)
+        device_list = []
+        if output:
+            pattern = r"^/dev/tty(AMA0|S0)$"
+            devices = output.splitlines()
+            device_list = [d for d in devices if re.search(pattern, d)]
+        return device_list
     except CalledProcessError as e:
         Logger.print_error("Unable to find a UART device!")
         Logger.print_error(e, prefix=False)

